game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -689,10 +690,18 @@ private void setCanNotTarget(final Set<UnitType> value) {
 
   public Set<UnitType> getCanNotTarget() {
     if (canNotTarget == null) {
-      canNotTarget =
-          new HashSet<>(
-              CollectionUtils.getMatches(
-                  getData().getUnitTypeList().getAllUnitTypes(), Matches.unitTypeIsAir()));
+      if (isSuicideOnAttack && isFirstStrike) { // Support isSuicide shortcut
```

Comment:
nit, not sure a casual reader will understand the comment to 'support isSuicide "shortcut"', they may wonder how it is a shortcut, and what 'isSuicide' is necessarily even referring to.

Reply:
Ok. Do you have an alternative suggestion?

Reply:
Some options:
- remove the comment
- expound on the comment and provide links/references for context
- bake the 'isSuicide' into matches. 
- Make 'CanTarget' a property of "UnitType", add explanations there

The last is probably my preferred approach followed by adding a specific matcher. I like the last approach as it has the best encapsulation and is the most easily tested (this section of code would be great to have tests for, it would make review easier and would lend more confidence this is working as intended).

Is there a legit timeline of when we can retrofit all maps to not have 'isSuicide'? IMHO the bulk map update experiences we have had caused far more pain than they are worth, and we do not even have ownership of all maps, and there was a lot of pain there. Removing map features I don't think is feasible and is part of why I think it's important we add versioning to map syntax/formats. Regardless of your viewpoint there, having a 'Matches.isSuicide' is independent of it is an actual unit property and the matches is, like here, a composite property.

Reply:
So the issue with that direction is that we really want to isolate and reduce where isSuicide is used as otherwise its confusing as whether the property is still being used throughout the delegates (which it isn't as its now just setting other properties). If I saw a matches isSuicide then I'm going to think its reading that property unless I go and specifically read that method. The idea here is to keep all the isSuicide functionality in the unit attachment so that it isn't leaked to the other classes.

Reply:
Updated to remove comment.